### PR TITLE
link: replace getopts with clap

### PR DESCRIPTION
**Pull request by deantvv**
Related to #2126

### CODE REVIEW

File: src/uu/link/src/link.rs [hunk 1]
```diff
@@ -8,14 +8,42 @@
...
+    pub static FILE2: &str = "FILE2";
+}
+
+#[derive(Debug)]
+struct Options {
+    old: PathBuf,
+    new: PathBuf,
+}
+
+impl Options {
```
Comment (sylvestre) [inline src/uu/link/src/link.rs:30]: are you sure this is necessary?

File: src/uu/link/src/link.rs [hunk 2]
```diff
@@ -25,16 +53,24 @@ pub fn normalize_error_message(e: Error) -> String {
...
+        .arg(
+            Arg::with_name(options::FILE1)
+                .hidden(true)
+                .takes_value(true),
+        )
+        .arg(
+            Arg::with_name(options::FILE2)
+                .hidden(true)
+                .takes_value(true),
+        )
```
Comment (jhscheer) [inline src/uu/link/src/link.rs:70]: You can set `required(true)` then clap will handle the checking and you don't need `msg_wrong_number_of_arguments!` anymore.
Alternatively you could use only one `Arg::with_name(options::FILE)` with `.multiple(true)` and  `.max_values(2)` and `.min_values(2)`.
And then get the paths like this:
```rust
let files: Vec<String> = matches
    .values_of(option::FILE)
    .map(|v| v.map(ToString::to_string).collect())
    .unwrap_or_default();
    let old = Path::new(&files[0]);
    let new = Path::new(&files[1]);
```

File: src/uu/link/src/link.rs [hunk 3]
```diff
@@ -8,14 +8,42 @@
...
+        let new = match matches.value_of_os(options::FILE2) {
+            Some(s) => Path::new(s).to_owned(),
+            None => crash!(1, "{}", msg_wrong_number_of_arguments!(2)),
+        };
+        Options { old, new }
+    }
+}
+
+fn get_usage() -> String {
+    format!("{0} FILE1 FILE2\tor: {0} OPTION", executable!())
```
Comment (jhscheer) [inline src/uu/link/src/link.rs:45]: GNU `link` uses `\n` instead of `\t`

Comment (jhscheer) [reply src/uu/link/src/link.rs:45]: on a closer look, it looks like `uu_link` doesn't support `link OPTION`

File: src/uu/link/src/link.rs (hunk 1 above)
Comment (deantvv) [reply src/uu/link/src/link.rs]: No, I will remove this and use the technique @jhscheer suggested below.

File: src/uu/link/src/link.rs (hunk 2 above)
Comment (deantvv) [reply src/uu/link/src/link.rs:47]: Yes, I will change to this format.

File: src/uu/link/src/link.rs (hunk 3 above)
Comment (deantvv) [reply src/uu/link/src/link.rs]: `link OPTION` is now removed.

**Comment by sylvestre** [comment]
perfect!
if you have a chance, you could add a test to check that it fails with 1 or 3 arguments

**Pull request merged**